src/http_connection.cpp

```diff
@@ -373,8 +373,11 @@ void http_connection::start(std::string const& hostname, int port
 #if defined TORRENT_ASIO_DEBUGGING
 			add_outstanding_async("http_connection::on_i2p_resolve");
```

Comment:
In that case I'd have to split *on_i2p_resolve* with the case when I explicitly add a peer. What I have indeed breaks semantics, but a matching completion is called in *on_i2p_resolve*. Shall I split that function then?